services/web/server/src/simcore_service_webserver/projects/_metadata_db.py

```diff
@@ -39,18 +58,52 @@ async def get_project_metadata(engine: Engine, project_uuid: ProjectID) -> Metad
         return parse_obj_as(MetadataDict, metadata.custom or {})
 
 
+_NIL_NODE_UUID: Final[NodeID] = NodeID(int=0)
+
+
 async def set_project_metadata(
     engine: Engine,
     project_uuid: ProjectID,
     custom_metadata: MetadataDict,
+    parent_project_uuid: ProjectID | None,
+    parent_node_id: NodeID | None,
 ) -> MetadataDict:
     """
     Raises:
         ProjectNotFoundError
+        NodeNotFoundError
         ValidationError: illegal metadata format in the database
     """
+    if parent_node_id is None and (parent_node_idstr := custom_metadata.get("node_id")):
+        # NOTE: backward compatibility with S4l old client
+        parent_node_id = parse_obj_as(NodeID, parent_node_idstr)
+
     async with _acquire_and_handle(engine, project_uuid) as connection:
+        if parent_node_id == _NIL_NODE_UUID:
+            parent_node_id = None
+        if parent_node_id and parent_project_uuid is None:
+            # let's try to get the parent project UUID
+            try:
+                parent_project_uuid = (
+                    await ProjectNodesRepo.get_project_id_from_node_id(
+                        connection, node_id=parent_node_id
+                    )
+                )
+
+            except ProjectNodesNodeNotFound as err:
+                raise DBProjectNodeParentNotFoundError((None, parent_node_id)) from err
+            except ProjectNodesNonUniqueNodeFoundError as err:
+                msg = "missing parent project id"
+                raise ProjectInvalidUsageError(msg) from err
```

Comment:
Lets talk about exceptions offline ...  I think we can optimize how we use exceptions